src/object/cli_obj.c

```diff
@@ -1430,37 +1426,60 @@ obj_iod_valid(unsigned int nr, daos_iod_t *iods, bool update)
 		switch (iods[i].iod_type) {
 		default:
 			D_ERROR("Unknown iod type=%d\n", iods[i].iod_type);
-			return false;
+			return -DER_INVAL;
 
 		case DAOS_IOD_NONE:
 			if (!iods[i].iod_recxs && iods[i].iod_nr == 0)
 				continue;
 
 			D_ERROR("IOD_NONE ignores value iod_nr=%d, recx=%p\n",
 				 iods[i].iod_nr, iods[i].iod_recxs);
-			return false;
+			return -DER_INVAL;
 
 		case DAOS_IOD_ARRAY:
 			if (!update && iods[i].iod_nr == 0) /* size query */
```

Comment:
hmm, why this is iod_nr?  it seems iod_size = 0 should be size query?

Reply:
that is original code.
for size query not must to set zero iod_size, just checked in srv_obj.c it checks it by "size_fetch = (!rma && orw->orw_sgls.ca_arrays == NULL);", so can be combined with NULL sgl case. I'll change it a little bit.